File: www/framework/Setter.js
//-------------------------------------------------------------------------------------------------------
// Copyright (C) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
//-------------------------------------------------------------------------------------------------------

//
// Applies a value to a property in a Style.
//
function Setter() {
    ace.NativeObject.call(this, "Windows.UI.Xaml.Setter");
};

// Inheritance
Setter.prototype = Object.create(ace.NativeObject.prototype);

Setter.prototype.getProperty = function () { return this.get("Setter.Property"); };
Setter.prototype.setProperty = function (propertyName) { this.set("Setter.Property", propertyName); };

Setter.prototype.getValue = function () { return this.get("Setter.Value"); };
Setter.prototype.setValue = function (value) { this.set("Setter.Value", value); };

module.exports = Setter;

File: src/android/framework/Setter.java
//-------------------------------------------------------------------------------------------------------
// Copyright (C) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
//-------------------------------------------------------------------------------------------------------
package Windows.UI.Xaml;

public class Setter implements Windows.UI.Xaml.Controls.IHaveProperties {
	public Setter(android.content.Context context) {
	}

	// IHaveProperties.setProperty
	public void setProperty(String propertyName, Object propertyValue)
	{
        if (propertyName.endsWith(".Property") || propertyName.endsWith(".Value")) {
            // Ignore. These are handled on the managed side.
        }
        else {
            throw new RuntimeException("Unhandled property for " + this.getClass().getSimpleName() + ": " + propertyName);
        }
	}
}
